### DISCUSSION TITLE
Apollo GraphQL and prisma.$use middleware

### DISCUSSION THREAD

**Discussion by darkodavidovic** [prisma/prisma#15955] [Q&A]
Hello... Can someone explain where to place prisma.$use(... in my Apollo GraphQL setup?
I have infinite loop of prisma.$use wherever i place prisma middleware and once i call my query from graph ql, prisma starts logging 50+ log messages in my terminal instead once for that specific model and action. In express without graph ql, prisma.$use works ok. Any example how to use it inside Apollo GraphQL? Thanks...

**Comment by austincrim** [comment]
Hey @darkodavidovic 👋,

Could you share an example (with code) of how you are using `prisma.$use` and Apollo?

**Reply by darkodavidovic** [reply]
Its right above ApolloServer constructor. Is any right place to do that in a better way?

```
resource.db.prisma.$use(async (params, next) => {
		console.log("middleware: ", params)
		if (params.model === 'User' && params.action === 'delete') {
			// code on delete
		}
		return await next(params)
	})

	const server = new ApolloServer({
		schema,
		context: createContext(resource.db.prisma),
		cache: 'bounded',
		plugins: [
			ApolloServerPluginDrainHttpServer({ httpServer }),
			ApolloServerPluginLandingPageLocalDefault({ embed: true }),
			{
				async serverWillStart() {
					return {
						async drainServer() {
							await serverCleanup.dispose();
						},
					};
				},
			},
		],
		introspection: true
	});

```

**Reply by austincrim** [reply]
I don't see anything obvious that's wrong in that snippet. I think I will need a minimal reproduction repo in order to continue reproduction.

**Reply by anitacung** [reply]
👋 Hi @darkodavidovic, 

I'm going through pending GitHub Discussions to check if you need any help. I also noticed that you opened another discussion about ["How to trigger some event after model is updated in Prisma?"](https://github.com/prisma/prisma/discussions/15982) and it has been resolved. Is this current discussion also resolved? If there's anything we can help you with, we'd love to get you in touch with our Product team.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
